test(client): fully cover client module

0ubbe opened the pull request:
## Reasoning 💡


<img width="1215" alt="Screenshot 2021-07-16 at 21 12 40" src="https://user-images.githubusercontent.com/5938217/125997606-ddc43116-ea92-4d05-b626-346bf2cdf4d8.png">


Final tests for `<SessionProvider />` and `useSession()` 🏋🏻‍♀️

## Checklist 🧢

<!-- Feel free cross items ( like this `~[] item~` ) if they're irrelevant to your changes.

To check an item, place an `x` in the box like so: `- [x] Documentation`. -->

- [ ] ~~Documentation~~
- [x] Tests
- [ ] ~~Ready to be merged~~

<!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->

## Affected issues 🎟

None

0ubbe committed:
test(client-provider): fix flaky test

0ubbe committed:
wip

0ubbe committed:
Merge 'origin/next' into 'test/client-tests'

0ubbe committed:
test(client-provider): verify more use-cases

0ubbe committed:
Merge "next" into "test/client-tests"

ndom91 reviewed (approved):
LGTM 👍 

I didn't follow the thread 100%, on mobile currently, but I saw your TODO, just curious why can't we clean __NEXTAUTH between test runs?

0ubbe committed:
Merge "next" into "test/client-tests"

0ubbe committed:
test(client): programmatic session refetch

0ubbe committed:
test(client): further coverage

balazsorban44 commented:
Small comment I noticed that `dist/client/__tests__` are also getting in the package published to npm. we should have a look.

don't see why those are generated at all, we only run tests, if not possible to not generate them, should we just delete after a test run?

0ubbe committed:
test(client): `stateTime` + `refetchInterval`

balazsorban44 committed:
Merge branch 'next' into test/client-tests

0ubbe committed:
refactor(client): test insights

• remove `staleTime`
• finish client tests

0ubbe committed:
refactor: unused variable

balazsorban44 committed:
chore: revert `package-lock.json` to  v2

0ubbe committed:
refactor: pair-review suggestions

• Bring back `staleTime` (remove it on an up-coming PR)
• Refine failed session flow
